frontend/packages/otr/src/components/publicInterpreter/listing/row/LanguagePairs.tsx

```diff
@@ -0,0 +1,78 @@
+import { useCallback } from 'react';
+import { Text } from 'shared/components';
+
+import { useKoodistoLanguagesTranslation } from 'configs/i18n';
+import { useAppSelector } from 'configs/redux';
+import { LanguagePair } from 'interfaces/languagePair';
+import { publicInterpretersSelector } from 'redux/selectors/publicInterpreter';
+import { QualificationUtils } from 'utils/qualifications';
+
+export const LanguagePairs = ({
+  languagePairs,
+}: {
+  languagePairs: Array<LanguagePair>;
+}) => {
+  const translateLanguage = useKoodistoLanguagesTranslation();
+
+  const {
+    filters: { fromLang, toLang },
+  } = useAppSelector(publicInterpretersSelector);
+
+  const sortAndRenderLanguagePairs = useCallback(
+    (
+      languagePairs: Array<LanguagePair>,
+      classes: string,
+      indexingStart: number,
+      ariaHidden?: boolean
+    ) => {
+      return languagePairs
+        .map((lp) =>
+          QualificationUtils.getLanguagePairLocalisation(lp, translateLanguage)
+        )
+        .sort()
+        .map((localisation, i) => (
+          <Text
+            key={i + indexingStart}
+            className={`${classes}`}
+            aria-hidden={ariaHidden}
+          >
+            {localisation}
+          </Text>
+        ));
+    },
+    [translateLanguage]
+  );
+
+  const groupLanguagePairsByFilters = () => {
+    if (fromLang && toLang) {
+      const matches = languagePairs.filter(
+        ({ from, to }) => from === fromLang && to === toLang
```

Comment:
OTR:ssä nämä pätevyydethän oli kaksisuuntaisia, tämä pitäis vissiin ottaa huomioon myös näitä matches- ja rest-listoja laskiessa.

Reply:
Tähän taitaakin löytyä jo valmiiksi `QualificationUtils.languagePairMatchesLangFilters`